--- libmp3splt-0.6.1a/src/plugins.c.wiget	2011-03-15 11:33:11.704147893 +0100
+++ libmp3splt-0.6.1a/src/plugins.c	2011-03-15 11:33:51.832156823 +0100
@@ -103,8 +103,8 @@ static int splt_p_filter_plugin_files(co
 
 #ifndef __WIN32__
         //gnu/linux .so.0
-        p_end = strstr(file, ".so.0");
-        if (p_end != NULL && (p_start == p_end) && (*(p_end+5) == '\0'))
+        p_end = strstr(file, ".so");
+        if (p_end != NULL && (p_start == p_end) && (*(p_end+3) == '\0'))
         {
           return 1;
         }
--- libmp3splt-0.6.1a/plugins/Makefile.am.wiget	2011-03-15 11:34:11.244527716 +0100
+++ libmp3splt-0.6.1a/plugins/Makefile.am	2011-03-15 11:34:41.346369612 +0100
@@ -5,7 +5,7 @@ plugin_LTLIBRARIES =
 #ccommon_LDFLAGS = -module -export-dynamic -avoid-version
 common_CPPFLAGS = -I$(top_srcdir)/include/libmp3splt -I$(top_srcdir)/src
 common_LDFLAGS = -L$(top_builddir)/src -L$(top_builddir)/src/.libs -L/lib \
-   -no-undefined -lm -lmp3splt @LIBINTL@
+   -no-undefined -avoid-version -lm -lmp3splt @LIBINTL@
 
 if WIN32
 common_LDFLAGS += -lz -lws2_32 -lintl
